syntax/grammar

```diff
@@ -0,0 +1,52 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl "," decl_list | epsilon
+
+type: i64 | i32 | i16 | i8 | i1 | f32 | f64 | str | '[' type ']' | void | NAME
```

Comment:
What do you mean by "array types are their own trait-able type"? Do you mean to say that putting a type `foo` in brackets makes it an array of type `foo`? If so it is accurately represented.
What about more complex data structures like dictionaries? Do they have their own type signature?